### DISCUSSION TITLE
Error running application NoClassDefFoundError

### DISCUSSION THREAD

**Discussion by joseoquendo** [primefaces/community#2745] [PrimeFaces] [answered]
Hello, I really need your help. My application is currently working fine with Java 7, Primefaces 3.5, JSF 2.1 and Glassfish 3.1.1. I updated the versions to Java 8, Primefaces 5.0, JSF 2.2 and Glassfish 4.1.2. However, when I try to run the application I get the following error:

Error occurred during deployment: Exception while loading the app : java.lang.IllegalStateException: ContainerBase.addChild: start: org.apache.catalina.LifecycleException: java.lang.RuntimeException: java.lang.NoClassDefFoundError: Lorg/primefaces/model/MenuModel;. Please see server.log for more details.

Thank you in advance for any help you can provide.

**Comment by melloware** [comment]
Hmmm what theme are you using if any?  This guy had the same issue and it turned out to be his Theme.JAR had the old model: https://forum.primefaces.org/viewtopic.php?t=68638

**Comment by joseoquendo** [comment]
I'm using the start theme, very old

**Reply by melloware** [reply]
Can you crack open its JAR and look for Java code? I am not familiar with start theme.

**Comment by joseoquendo** [comment]
JAR file

pom.xml
![Captura de tela de 2024-09-25 14-55-19](https://github.com/user-attachments/assets/931ef8eb-91ed-402a-93aa-345c19e85ece)

pom.properties
![Captura de tela de 2024-09-25 14-56-18](https://github.com/user-attachments/assets/a8b1eaf5-2213-42c8-94c6-c21a9c88d77d)

**Reply by melloware** [reply]
Yep no java code in those old themes.  All I can say is you have a class conflict somewhere in your classpath.

**Comment by joseoquendo** [comment]
The worst thing is that we have had this error for some time now and we have not found where this conflict is.

**Reply by melloware** [reply]
Are you using any other Company JARS that may have PF code?  It has to be somewhere.

**Comment by joseoquendo** [comment, marked as answer]
From primefaces we are only using 2, primefaces-5.0.jar and start-1.0.10.jar which is from the theme. Before I was using primefaces-3.5.jar and start-1.0.2.jar

**Reply by melloware** [reply]
Yep and you are on ancient versions so its really impossible to help.  But the error is definitely a No Class Def Found so someting is wrong have you tried a different PF Jar like 6.0?

**Reply by melloware** [reply]
Another guy with the same issue trying the same upgrade: https://github.com/orgs/primefaces/discussions/2727

**Reply by joseoquendo** [reply]
Yes, I talked to him, but it still hasn't been resolved.

**Reply by melloware** [reply]
Ok I will close that other thread if you guys are working together?

**Reply by joseoquendo** [reply]
Ok, we will continue the fight to find the problem, thanks for help

**Reply by melloware** [reply]
I am actually wonder because I know GlassFish and Payara do aggressive scanning of the faces-config.xml and primefaces.taglib.xml if somehow something is wrong there and Glassfish is loading it.  That is why I was suggesting trying a newer PF like 6.0 etc.
